### ISSUE TITLE
Dropdown css: 1px shift of the dropdown in the classic notebook

### ISSUE THREAD

**Issue by SylvainCorlay** [ipython/ipywidgets#1029]
The rendering seems to be a little off in the case of the classic notebook.

**Comment by jasongrout** [comment]
Can you post a screenshot?

**Comment by SylvainCorlay** [comment]
There you go.

Classic:

<img width="588" alt="screen shot 2017-01-07 at 11 03 58 am" src="https://cloud.githubusercontent.com/assets/2397974/21740953/70c1d85a-d4c9-11e6-8789-20b49e2b9dd9.png">

Lab

<img width="639" alt="screen shot 2017-01-07 at 11 04 53 am" src="https://cloud.githubusercontent.com/assets/2397974/21740954/7821c088-d4c9-11e6-8ddb-b0cca72de563.png">

**Comment by jdetle** [comment]
I can take care of this if @jasongrout hasn't already started on it.

**Comment by SylvainCorlay** [comment]
@jdetle 👍 . I don't that anyone has tackled this.

**Comment by jdetle** [comment]
Seems like there are two issues here, when the classic notebook has rendered an output above, the dropdown menu will (as of now) inexplicably render a bottom border beneath the 2 as seen in your example, but will not if there is no output area being rendered above the cell rendering the dropdown. In addition, there is a larger gap in the notebook than in the lab.

**Comment by SylvainCorlay** [comment]
The bottom border is always there but should be hidden by the dropdown.

**Comment by jdetle** [comment]
Right, and in some cases it isn't. In the picture you gave it is in JLab but not in the Classic Notebook.

**Comment by jdetle** [comment]
Two issues I found:
The fix for the border issue is changing 
```
ul.widget-dropdown-droplist.below {
    border-top: none;
    transform: translateY(calc(-1 * var(--jp-border-width)));
}
```
to 
```
ul.widget-dropdown-droplist.below {
    border-top: none;
    transform: translateY(calc(-2 * var(--jp-border-width)));
}
```

the different pixel difference comes from a different DOM tree and css associated with JLab that isn't associated with the widgets or the notebook.

<img width="441" alt="screen shot 2017-01-14 at 10 52 24 am" src="https://cloud.githubusercontent.com/assets/10944582/21957275/e50d7dbc-da47-11e6-9196-a32c562378e2.png">

<img width="494" alt="screen shot 2017-01-14 at 10 51 54 am" src="https://cloud.githubusercontent.com/assets/10944582/21957278/e97ee2c8-da47-11e6-9b37-bf95502d621c.png">

JP-Output:firstChild {
   margin-top: 5px;
}
gives this spacing in the JLab rendering but no such Output div exists in the dom of the notebook, (at least as far I can tell.)

**Comment by SylvainCorlay** [comment]
@jdetle it seems that the shift comes from the rounding of the computed client rect. I have proposed a fix in #1040 .

**Comment by jdetle** [comment]
Ah, that makes more sense. Thank you.
